src/shared/bpf-program.c

```diff
@@ -254,3 +254,29 @@ int bpf_map_lookup_element(int fd, const void *key, void *value) {
 
         return 0;
 }
+
+int bpf_obj_get_info_by_fd(int prog_fd, void *info, uint32_t *info_len) {
```

Comment:
Renamed. Now `bpf_object_get_info` is a wrapper for `bpf_obj_get_info` from LIBBPF_API returning `-ENOTSUP` if libbpf is not found.
Wdyt about the approach I've used in this PR: for new features avoid direct syscalls. Instead, guard LIBBPF_API calls by `#if HAVE_LIBBPF` macro. This will not affect legacy features, until they are explicitly switched to code compiled from sources.